build-logic/api-extractor/src/main/java/org/gradle/internal/tools/api/ApiMemberWriter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.tools.api;
+
+import org.gradle.internal.tools.api.impl.AnnotationMember;
+import org.gradle.internal.tools.api.impl.AnnotationValue;
+import org.gradle.internal.tools.api.impl.ClassMember;
+import org.gradle.internal.tools.api.impl.FieldMember;
+import org.gradle.internal.tools.api.impl.InnerClassMember;
+import org.gradle.internal.tools.api.impl.MethodMember;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.ModuleVisitor;
+
+import java.util.Set;
+
+public interface ApiMemberWriter {
```

Comment:
I think this file is worthy of _some_ javadoc

Reply:
Added some javadoc